contrib/educational_decoder/harness.c

```diff
@@ -0,0 +1,93 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "zstd_decompress.h"
+
+typedef unsigned char u8;
+
+// There's no good way to determine output size without decompressing
+// For this example assume we'll never decompress at a ratio larger than 16
+#define MAX_COMPRESSION_RATIO (16)
```

Comment:
I would recommend usage of `ZSTD_getDecompressedSize()` to determine output buffer size
(or any other equivalent of your liking).

You can still have a fallback using `MAX_COMPRESSION_RATIO`, since in some cases (streaming pipe), decompressed size is unknown (not stored in the header).
In which case, use a relevant warning message to inform users about what's happening.